mapproxy/service/templates/demo/tms_demo.html

```diff
@@ -35,11 +39,17 @@
             ol.proj.proj4.register(proj4);
         }
 
+        // Define TMS as specialized XYZ service: origin lower-left and may have custom grid
         const source = new ol.source.XYZ({
-            url: '../tms/{{"/".join(layer.md["name_path"])}}/{z}/{x}/{-y}.png',
-            opaque: transparent === "true" ? false : true,
+            url: '../tms/1.0.0/{{"/".join(layer.md["name_path"])}}/{z}/{x}/{-y}.' + format,
+            opaque: transparent,
             projection: "{{srs}}",
-            maxResolution: {{resolutions[0]}}
+            maxResolution: {{resolutions[0]}},
+            tileGrid: new  ol.tilegrid.TileGrid({
+               tileSize: 256,
```

Comment:
@justb4 should this be using the `tileSize` variable defined on line 26?

Reply:
Actually I think line 26 should be along these lines (assuming `json` has been imported):
```
        const tileSize = {{json.dumps(layer.grid.tile_size)}};  // TODO allow custom e.g. Retina tilesizes
```
While testing the value of `tileSize` was `"(256, 256)"` for the demo OSM layer.

Reply:
@walkermatt yes,  `tileSize` (and maybe `origin`) was not present even in the pre-OL7 version. Had to figure out how to convert from what looks like a stringified Python `tuple` to a JS `array`, but you nailed it, thanks! Edit upcoming.

Reply:
and the [WMTS demo HTML template](https://github.com/mapproxy/mapproxy/blob/master/mapproxy/service/templates/demo/wmts_demo.html#L55) also lacks the tile size, will add there as well and slightly change PR title/desc. (found that while working on HQ 512x512 tiles).

Reply:
Also fixed tile sizes in WMTS and TMS. 

Only remaining item is `origin`: as TMS is emulated using the class `ol.source.XYZ` with flipped `y` `{-y}`, I wondered if `origin` should be set explicit similar to WMTS, like: ` origin: ol.extent.getBottomLeft(grid_extent)`  But that had reverse effects. IMHO origin of TMS is implicit (lower-left) and MapProxy assumes that, so independent how the internal tile-grid is configured  (`sw` or `nw`), the proper tiles are fetched.  I keep it this way (no explicit `origin` for TMS).

So al in all, think good to go, pending reviews.